hcloud/exp/kit/envutils/env_test.go

```diff
@@ -0,0 +1,104 @@
+package envutils
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func writeTmpFile(t *testing.T, tmp string, filename string, content string) string {
+	filepath := path.Join(tmp, filename)
+
+	err := os.WriteFile(filepath, []byte(content), 0644)
+	require.NoError(t, err)
+
+	return filepath
+}
+func TestLookupWithFile(t *testing.T) {
+	testCases := []struct {
+		name  string
+		setup func(t *testing.T, tmpDir string)
+		want  func(t *testing.T, value string, err error)
+	}{
+		{
+			name:  "without any environment",
+			setup: func(t *testing.T, tmpDir string) {},
+			want: func(t *testing.T, value string, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, "", value)
+			},
+		},
+		{
+			name: "value from environment",
+			setup: func(t *testing.T, tmpDir string) {
+				t.Setenv("CONFIG", "value")
+
+				// Test for precedence
+				filepath := writeTmpFile(t, tmpDir, "config", "content")
+				t.Setenv("CONFIG_FILE", filepath)
+			},
+			want: func(t *testing.T, value string, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, "value", value)
+			},
+		},
+		{
+			name: "empty value from environment",
+			setup: func(t *testing.T, tmpDir string) {
+				t.Setenv("CONFIG", "")
+
+				// Test for precedence
+				filepath := writeTmpFile(t, tmpDir, "config", "content")
+				t.Setenv("CONFIG_FILE", filepath)
```

Comment:
Do we agree that if an env var is present but empty, we do not check for the file? This feels like the less supprising.

Reply:
Agree